projects/shared/src/lib/contentlist/contentlist.component.ts

```diff
@@ -1,15 +1,44 @@
-import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { OnInit } from '@angular/core';
+import { Component, Input, Output, EventEmitter, ChangeDetectorRef } from '@angular/core';
+import { ContentListItem } from './content-list-item';
 
 @Component({
   selector: 'lib-contentlist',
   templateUrl: './contentlist.component.html',
   styleUrls: ['./contentlist.component.css']
 })
-export class ContentlistComponent {
-  @Input() contentList;
+export class ContentlistComponent implements OnInit {
+  @Input() contentList: Array<any> = [];
   @Output() contentSelect = new EventEmitter<any>();
 
+  public contents: Array<ContentListItem>
+  ngOnInit() {
+
+    this.contents = this.contentList.map(content => {
```

Comment:
I was expecting something like this
https://dev.to/codeprototype/safely-parsing-json-to-a-typescript-interface-3lkj

So any change in the interface class requires change in the .convert function only.


Component will ve completely object properties agnostic.